File: BOTAIML.VisionBot.Monitoring.Web/ClientApp/src/app/api/models/sub-center-area-log.ts
/* tslint:disable */
/* eslint-disable */
import { AreaName } from './area-name';
import { DoorStatus } from './door-status';
import { Person } from './person';
import { PowerStatus } from './power-status';
import { TrackingStatus } from './tracking-status';
export interface SubCenterAreaLog {
  cameraId?: number;
  doorStatus?: DoorStatus;
  id?: number;
  isAlertRequired?: boolean;
  isAlertSent?: boolean;
  name: AreaName;
  person?: Person;
  personId?: null | number;
  powerStatus?: PowerStatus;
  probability?: null | number;
  status?: TrackingStatus;
  time: string;
}

File: BOTAIML.VisionBot.Monitoring.Web/Models/SubCenterAreaLog.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BOTAIML.VisionBot.Monitoring.Web.Models
{
    public class SubCenterAreaLog
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        public AreaName Name { get; set; }
        public DoorStatus? DoorStatus { get; set; }

        [ForeignKey(nameof(Person))]
        public int? PersonId { get; set; }

        [Required]
        public DateTime Time { get; set; }
        public TrackingStatus? Status { get; set; }
        public float? Probability { get; set; }
        public PowerStatus? PowerStatus { get; set; }
        public int CameraId { get; set; }
        public bool IsAlertRequired { get; set; }

        public bool IsAlertSent { get; set; }

        public virtual Person Person { get; set; }

    }
}